core/chains/evm/txmgr/orm.go

```diff
@@ -540,3 +575,182 @@ WHERE eth_tx_attempts.state = 'in_progress' AND eth_txes.from_address = $1 AND e
+	}, pg.OptReadOnlyTx())
+	if errors.Is(err, sql.ErrNoRows) {
+		return nil, nil
+	}
+	return
+}
+
+func updateEthTxAttemptUnbroadcast(q pg.Queryer, attempt EthTxAttempt) error {
+	if attempt.State != EthTxAttemptBroadcast {
+		return errors.New("expected eth_tx_attempt to be broadcast")
+	}
+	_, err := q.Exec(`UPDATE eth_tx_attempts SET broadcast_before_block_num = NULL, state = 'in_progress' WHERE id = $1`, attempt.ID)
+	return errors.Wrap(err, "updateEthTxAttemptUnbroadcast failed")
+}
+
+func updateEthTxUnconfirm(q pg.Queryer, etx EthTx) error {
+	if etx.State != EthTxConfirmed {
+		return errors.New("expected eth_tx state to be confirmed")
+	}
+	_, err := q.Exec(`UPDATE eth_txes SET state = 'unconfirmed' WHERE id = $1`, etx.ID)
+	return errors.Wrap(err, "updateEthTxUnconfirm failed")
+}
+
+func deleteEthReceipts(q pg.Queryer, etxID int64) (err error) {
+	_, err = q.Exec(`
+DELETE FROM eth_receipts
+USING eth_tx_attempts
+WHERE eth_receipts.tx_hash = eth_tx_attempts.hash
+AND eth_tx_attempts.eth_tx_id = $1
+	`, etxID)
+	return errors.Wrap(err, "deleteEthReceipts failed")
+}
+
+func (o *orm) UpdateEthTxForRebroadcast(etx EthTx, etxAttempt EthTxAttempt) error {
+	return o.q.Transaction(func(tx pg.Queryer) error {
+		if err := deleteEthReceipts(tx, etx.ID); err != nil {
+			return errors.Wrapf(err, "deleteEthReceipts failed for etx %v", etx.ID)
+		}
+		if err := updateEthTxUnconfirm(tx, etx); err != nil {
+			return errors.Wrapf(err, "updateEthTxUnconfirm failed for etx %v", etx.ID)
+		}
+		return updateEthTxAttemptUnbroadcast(tx, etxAttempt)
+	})
+}
+
+func (o *orm) FindTransactionsConfirmedInBlockRange(highBlockNumber, lowBlockNumber int64, chainID big.Int) (etxs []*EthTx, err error) {
+	err = o.q.Transaction(func(tx pg.Queryer) error {
+		err = tx.Select(&etxs, `
+SELECT DISTINCT eth_txes.* FROM eth_txes
+INNER JOIN eth_tx_attempts ON eth_txes.id = eth_tx_attempts.eth_tx_id AND eth_tx_attempts.state = 'broadcast'
+INNER JOIN eth_receipts ON eth_receipts.tx_hash = eth_tx_attempts.hash
+WHERE eth_txes.state IN ('confirmed', 'confirmed_missing_receipt') AND block_number BETWEEN $1 AND $2 AND evm_chain_id = $3
+ORDER BY nonce ASC
+`, lowBlockNumber, highBlockNumber, chainID.String())
+		if err != nil {
+			return errors.Wrap(err, "FindTransactionsConfirmedInBlockRange failed to load eth_txes")
+		}
+		if err = loadEthTxesAttempts(tx, etxs); err != nil {
+			return errors.Wrap(err, "FindTransactionsConfirmedInBlockRange failed to load eth_tx_attempts")
+		}
+		err = loadEthTxesAttemptsReceipts(tx, etxs)
+		return errors.Wrap(err, "FindTransactionsConfirmedInBlockRange failed to load eth_receipts")
+	}, pg.OptReadOnlyTx())
+	return etxs, errors.Wrap(err, "FindTransactionsConfirmedInBlockRange failed")
+}
+
+func (o *orm) saveAttemptWithNewState(q pg.Queryer, timeout time.Duration, attempt EthTxAttempt, broadcastAt time.Time) error {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
```

Comment:
Should be pegged to orm-global context so it shuts down with the orm

Reply:
This method `saveAttemptWithNewState` actually could be removed from the orm struct ... it should be passed in a query. But that being said all usages of this function ultimately derive from the the orm queryer `o.q`.

I will remove it as a method to the orm struct and make it a vanilla private function